src/Semaphore.sol

```diff
@@ -164,4 +187,29 @@ contract Semaphore is IWorldID, SemaphoreCore, Verifier, SemaphoreGroups {
 
         manager = newManager;
     }
+    ///////////////////////////////////////////////////////////////////////////////
+    ///                          VIEW FUNCTIONS                                ///
+    //////////////////////////////////////////////////////////////////////////////
+
+    /// @notice Checks if a given root value is valid and has been added to the root history
+    /// @param root  The root of a given identity group
+    /// @param groupId The id of the group
+    function checkValidRoot(uint256 root, uint256 groupId) public view returns (bool) {
```

Comment:
Order of variables is more natural as `groupId`, `root`.

Reply:
Agreed!